cns/service/main.go

```diff
@@ -1245,3 +1249,112 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 
 	return nil
 }
+
+func SetMellanoxRegKey() {
+	var mellanoxSearchString = "*Mellanox*"
+	var priorityVLANTagIdentifier = "*PriorityVLANTag"
+	var registryKeyPrefix = "HKLM:\\System\\CurrentControlSet\\Control\\Class\\"
+
+	//get adapter name
+	cmd := fmt.Sprintf("Get-NetAdapter | Where-Object { $_.InterfaceDescription -like \"%s\" } | Select-Object -ExpandProperty Name", mellanoxSearchString)
+	adapatername, err := platform.ExecutePowershellCommand(cmd)
+
+	if err != nil {
+		log.Printf("Error while executing pwsl command to get net adapter list: %v", err)
+		return
+	}
+	if adapatername == "" {
+		log.Printf("No Network adapter found with \"%s\" in description", mellanoxSearchString)
+		return
+	}
+
+	log.Printf("Name of Mellanox adapter : %v", adapatername)
+
+	//Find if adapter with Priority vlan tag (version 4 or up) or not(version 3)
+	cmd = fmt.Sprintf("Get-NetAdapterAdvancedProperty | Where-Object { $_.RegistryKeyword -like \"%s\" -and $_.Name -eq \"%s\" } | Select-Object -ExpandProperty Name", priorityVLANTagIdentifier, adapatername)
+	log.Printf("%v", cmd)
+	adapterNamewithVLANTag, err := platform.ExecutePowershellCommand(cmd)
```

Comment:
adapter name having vlan tag in advance property. So for version 4 it will return name while for version 3 it will be blank since no *priorityvlantag reg key found